src/compiler/codegen/events.js

```diff
@@ -16,6 +16,18 @@ const keyCodes: { [key: string]: number | Array<number> } = {
   'delete': [8, 46]
 }
 
+const keyNames: { [key: string]: string | Array<string> } = {
```

Comment:
Why not just move the keyNames and keyCodes to runtime instead of in the compiler? Since it leads sooo much repetition and should reduce the code size.

Reply:
Actually in my first version, I did define `keyNames` in `check-keycodes.js`. After that I realize this would increase much runtime bundle size, so I got it moved into compiler...

Reply:
I guess the reason `keyCodes` stays in compiler is that it simply generate numbers rather strings. Agreed for moving `keyNames` into runtime.🙂

Reply:
Considering all compiled templates size together, it could reasonable to move them into runtime. BTW, seems it has not that much different to deal with keyCodes and keyNames.